packages/block-editor/src/hooks/auto-inserting-blocks.js

```diff
@@ -15,39 +15,32 @@ import { useDispatch, useSelect } from '@wordpress/data';
 import { InspectorControls } from '../components';
 import { store as blockEditorStore } from '../store';
 
-function AutoInsertingBlocksControl( props ) {
-	const { autoInsertedBlocksForCurrentBlock, groupedAutoInsertedBlocks } =
-		useSelect(
-			( select ) => {
-				const { getBlockTypes } = select( blocksStore );
-				const _autoInsertedBlocksForCurrentBlock =
-					getBlockTypes()?.filter(
-						( { autoInsert } ) =>
-							autoInsert && props.blockName in autoInsert
-					);
+function someBlock( blocks, predicate, getBlocks ) {
+	return blocks.some( ( block ) => {
+		const result = predicate( block );
+		if ( result ) {
+			return result;
+		}
 
-				// Group by block namespace (i.e. prefix before the slash).
-				const _groupedAutoInsertedBlocks =
-					_autoInsertedBlocksForCurrentBlock?.reduce(
-						( groups, block ) => {
-							const [ namespace ] = block.name.split( '/' );
-							if ( ! groups[ namespace ] ) {
-								groups[ namespace ] = [];
-							}
-							groups[ namespace ].push( block );
-							return groups;
-						},
-						{}
-					);
+		// A Template Part block's innerBlocks array always shows up as empty,
+		// so we have to manually load its inner blocks via getBlocks().
+		const innerBlocks =
+			block.name === 'core/template-part'
```

Comment:
Should it check if the block container is controlled instead? It might be more blocks that are configured this way.

Reply:
Good point! But it seems like we can eschew that problem altogether thanks to [your other suggestion](https://github.com/WordPress/gutenberg/pull/54024#discussion_r1309720885) 😄